### ISSUE TITLE
GoRun GoTest errors

### ISSUE THREAD

**Issue by Davincible** [ray-x/go.nvim#146]
I've been playing with `GoRun` and `GoTest` and noticed a few bugs;

~~* (GoTest) if you set a non-default runner, e.g. richgo, and run only `GoTest` without any args, it installs it, and runs the test. But next time around it has to install it again, and the binary is also not callable from anywhere else. So it only does a temporary install.~~ (bug on my side)
* (GoTest) if you pass in the `-F` flag, and have a non-default test runner. It does install it, but only after printing out an error about it not being found. Then opens the terminal prior to it being installed, and then never actually runs the tests.
* (Both) If you pass in the `-F` flag, and an error occurs, the floating term still opens up empty even though an error occurred. Would be nice to have a check that if an error occurred, don't open floating term anymore.

**Comment by Davincible** [comment]
Turns out some of the bugs previously reported were misconfigs on my side. Updated the comment. 

Side request; it would be nice if running `GoRun` on a `main.go` file or `main` package would automatically run that main file/package

**Comment by ray-x** [comment]
`-F` option allows you run in floaterm, without it it runs in background and send result to loclist. 
If you set `run_in_floaterm` to be true, you can omit `-F` option.
Pushed an update and hope it solves the defect you found.

**Comment by Davincible** [comment]
I get the same error as you GitHub tests
`E5108: Error executing lua ...share/nvim/site/pack/vendor/start/go.nvim/lua/go/env.lua:19: attempt to concatenate local 'val' (a userdata value)`

**Comment by Davincible** [comment]
if I change your append function to this, it works;
```lua
function M.append(env, val)
  local oldval = vfn.getenv(env)
  if val == vim.NIL or string.find(oldval, val) then
    return
  end
  local newval = oldval .. ":" .. val
  vfn.setenv(env, newval)
end

```
